File: randlib/randlib.lua
RANDLIB_MAX = 32767
INT_LIMIT = 2147483657
seed = 0
prevSeed = seed

local Randlib = {}

function NewRand()
	repeat
		seed = os.time()
	until seed ~= prevSeed
	prevSeed = seed
	return ((8253739 * seed + 2396403) % INT_LIMIT) % RANDLIB_MAX
end

function Randlib.Set_RANDLIB_MAX(value)
	RANDLIB_MAX = value
end

function Randlib.Randint(min, max)
	if min > max then
		max1 = max
		max = min
		min = max1
	end
	min = math.floor(min+0.5)
	max = math.floor(max+0.5)
	local value = min + (NewRand()%(max - min +1))
	return value
end

function Randlib.Randfloat(min, max)
	if min > max then
		max1 = max
		max = min
		min = max1
	end
	pre = Randlib.Randint(math.floor(min),math.floor(max)-1)
	post = NewRand()/10^ math.floor(math.log(RANDLIB_MAX, 10))
	return pre + post
end

function Randlib.Randchar(excludeWhitespaces)
	local c
	local whitespaces ={' ', '\t', '\n'}
	if not excludeWhitespaces then
		if Randlib.Randint(0,6) == 6 then
			c = whitespaces[Randlib.Randint(1,3)]
		else
			c = string.char(Randlib.Randint(33,126))
		end
	else
		c = string.char(Randlib.Randint(33,126))
	end
	return c
end

function Randlib.Randstring(length, excludeWhitespaces)
	local text = ""
	for i=0, length-1 do
		repeat
			c = Randlib.Randchar(excludeWhitespaces)
		until c ~= nil
		text = text .. c
	end
	return text
end

return Randlib 

File: randlib/randlib.dart
import "dart:math";
int seed = 0;
int prevSeed = seed;

int NewRand(int RANDLIB_MAX){
  while (seed == prevSeed){
    seed = DateTime.now().millisecondsSinceEpoch;
  }
  prevSeed = seed;
  int value =  (8253739 * seed + 2396403) % RANDLIB_MAX;
  return value;
}

int Randint(int min, int max, int RANDLIB_MAX){
  if (min > max){
    int max1 = max;
    max = min;
    min = max1;
  }
  int value = min + (NewRand(RANDLIB_MAX) % (max - min +1));
  return value;
}

double Randfloat(double min, double max, int RANDLIB_MAX){
  if (min > max){
    double max1 = max;
    max = min;
    min = max1;
  }
  double pre, post;
  pre = Randint(min.floor(), max.floor()-1, 32767).toDouble();
  post = NewRand(32767) / pow(10, (log(RANDLIB_MAX)/ln10).floor()+1);
  return pre + post;
}

bool Randbool(){
  return Randint(0,1,32767) == 1;
}

String Randchar(bool excludeWhitespaces){
  int ascii_code=0;
  var whitespaces = [9, 10, 32];
  if (!excludeWhitespaces && randint(0,6,32767) == 0){
    ascii_code = ascii_code = whitespaces[Randint(0,2,32767)];
  }
  else{
    ascii_code = Randint(33,126,32767);
  }
  return String.fromCharCode(ascii_code);
  
}

String Randstring(int length, bool excludeWhitespaces){
  String text = "";
  for (int i = 0; i < length; i++){
    text += Randchar(excludeWhitespaces);
  }
  return text;
}
